blocks/model.py

```diff
@@ -155,8 +155,12 @@ def __init__(self, outputs):
         for brick in bricks:
             if brick not in children and brick not in self.top_bricks:
                 self.top_bricks.append(brick)
-        if len(set(b.name for b in self.top_bricks)) < len(self.top_bricks):
-            raise ValueError("top bricks with the same name")
+        # check there is no occurence in top bricks name
+        names = Counter([b.name for b in self.top_bricks])
```

Comment:
This might be an easier way to get all the names that are not unique, instead of just one:
```python
names = set()
repeated_names = [brick.name for brick in self.top_bricks if brick.name in names or names.add(brick.name)]
```